drivers/stmpe811/include/stmpe811_params.h

```diff
@@ -67,6 +67,21 @@ static const stmpe811_params_t stmpe811_params[] =
     STMPE811_PARAMS
 };
 
+/**
+ * @brief   Default screen identifiers
+ */
+#ifndef STMPE811_PARAM_SCREEN_IDS
+#define STMPE811_PARAM_SCREEN_IDS       0
+#endif
+
+/**
+ * @brief   Configure screen identifiers
+ */
+static const uint8_t stmpe811_screen_ids[] =
```

Comment:
I strongly disagree with this as this is asking for out-of-bounds array reads. This array should either be always as long as the `stmpe811_params` array, or ID's should be autogenerated in a predictable way on start. The current enhancement makes it error prone as it is easy to extend the params struct and forget to extend this struct.

Reply:
How about we add an assert in the `auto_init_screen` function to check that both params and screen_ids arrays have the same size ?

Reply:
ping @bergzand

Reply:
Added an assert in each `auto_init_<driver>.c` to check the length of the params array is the same a the length of the ids array.

Reply:
I think the same handling is done currently for SAUL which is solved with an assert as well, so I think that is OK